de-lib/src/main/java/org/iplantc/de/client/services/impl/DiskResourceServiceFacadeImpl.java

```diff
@@ -1041,6 +1042,48 @@ public String downloadTemplate(String templateid) {
         return address;
     }
 
+    @Override
+    public void requestHTPathlistFile(HTPathListRequest request, DECallback<File> callback) {
+        StringBuilder address =
+                new StringBuilder(deProperties.getDataMgmtBaseUrl() + "path-list-creator?");
+        address.append("dest=" + URL.encodeQueryString(request.getDest()));
+        if(!Strings.isNullOrEmpty(request.getPattern())) {
+            address.append("&name-pattern=" + URL.encodeQueryString(request.getPattern()));
+        }
+        address.append("&folders-only=" + request.isFoldersOnly());
+        address.append("&recursive=" + true);
+        if (request.getInfoTypes() != null && request.getInfoTypes().size() > 0) {
+            for (String infotype : request.getInfoTypes()) {
+                address.append("&info-type=" + infotype);
+            }
+        }
+        Splittable s = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(request));
+        Splittable spath = StringQuoter.createSplittable();
+        s.get("paths").assign(spath,"paths");
```

Comment:
I guess I was thinking of something like this, where maybe the request can be encoded then decoded into a new `HasPaths` AutoBean: https://github.com/cyverse-de/ui/blob/2.13.0/de-lib/src/main/java/org/iplantc/de/client/services/impl/DiskResourceServiceFacadeImpl.java#L506

Or even just getting a new `HasPaths` from the factory and setting its paths from the request, then the new `HasPaths` can be encoded as the request body: https://github.com/cyverse-de/ui/blob/2.13.0/de-lib/src/main/java/org/iplantc/de/client/services/impl/DiskResourceServiceFacadeImpl.java#L602